src/Enum.php

```diff
@@ -144,26 +135,20 @@ public static function make($value): Enumerable
                 throw new InvalidIndexException($value, static::class);
             }
 
-            $name = array_combine(static::getIndices(), array_keys(static::resolve()))[$value];
-            $index = $value;
-            $value = array_search($index, static::toArray());
+            list($name, $index, $value) = static::resolveByIndex($value);
         } elseif (is_string($value)) {
-            if (method_exists(static::class, $value)) {
-                return forward_static_call(static::class.'::'.$value);
-            }
-
             if (static::isValidValue($value)) {
-                $index = static::toArray()[$value];
-                $name = array_combine(static::getValues(), array_keys(static::resolve()))[$value];
+                list($name, $index, $value) = static::resolveByValue($value);
             } elseif (static::isValidName($value)) {
```

Comment:
We could remove the need for an `else` by adding a function that returns an array with 3 elements.

Reply:
So combine them in a method `resolveByString()`?

Reply:
Sound like a plan! 👍